internal/pkg/cli/selector/selector.go

```diff
@@ -51,16 +60,24 @@ type WorkspaceSelect struct {
 	svcLister wsSvcLister
 }
 
+// DeploySelect is a service and environment selector from the deploy store.
+type DeploySelect struct {
+	*Select
+	deployStoreSvc deployStoreClient
+	svc            string
+	env            string
+}
+
 // NewSelect returns a selector that chooses applications or environments.
-func NewSelect(prompt Prompter, store *config.Store) *Select {
+func NewSelect(prompt Prompter, store appEnvLister) *Select {
```

Comment:
v nice thank you! I still find it super strange to have private interfaces in a constructor? Clients have noway of knowing how to implement this interface:

![Screen Shot 2020-07-13 at 9 38 53 AM](https://user-images.githubusercontent.com/879348/87329972-b9fe5400-c4ec-11ea-9353-8242c12d2936.png)

Here is what the godoc looks like for this fn, I think these interfaces should be public instead